### PR TITLE
mongosh 1.6.2 -> 1.8.0

### PR DESCRIPTION
**Pull request by undefined-moe**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
Build is failing on linux:
```
npm ERR! In file included from ../src/mongocrypt.cc:1:
npm ERR! ../src/mongocrypt.h:15:10: fatal error: mongocrypt/mongocrypt.h: No such file or directory
npm ERR!    15 | #include <mongocrypt/mongocrypt.h>
npm ERR!       |          ^~~~~~~~~~~~~~~~~~~~~~~~~
```

File: pkgs/development/tools/mongosh/default.nix [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-    inherit (stdenv.hostPlatform) system;
-  };
-in
-nodePackages.mongosh.override {
-  passthru.tests.version = testers.testVersion {
-    package = mongosh;
+buildNpmPackage rec {
+  pname = "mongosh";
+  version = "1.7.1";
+  src = fetchzip {
```
Comment (NickCao) [inline pkgs/development/tools/mongosh/default.nix:6]: Upstream does provide a lock file, please use `fetchFromGitHub`: https://github.com/mongodb-js/mongosh

File: pkgs/development/tools/mongosh/default.nix [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
+    url = "https://registry.npmjs.com/mongosh/-/mongosh-${version}.tgz";
+    hash = "sha256-LEV1nKuUP0TwB0eFm04VPZyEJOzvlNElzSg0QlkYGPI=";
   };
+  postPatch = "cp ${./package-lock.json} package-lock.json";
+  npmDepsHash = "sha256-Ho2lpA8dHkiDF8G4GkiEWFYCFnqOuqYx+ZMxxG8D8BU=";
+  dontNpmBuild = true;
+  npmFlags = ["--legacy-peer-deps" "--omit=optional"];
+  makeCacheWritable = true;
+  # NODE_OPTIONS = "--openssl-legacy-provider";
```
Comment (NickCao) [inline pkgs/development/tools/mongosh/default.nix:15]: Remove unused options, not commenting them out.

File: pkgs/development/tools/mongosh/default.nix (hunk 1 above)
Comment (undefined-moe) [reply pkgs/development/tools/mongosh/default.nix:6]: This lockfile is not for mongosh package, it contains all dependencies used in the whole workspace, which contains lots of unused packages.

Comment (NickCao) [reply pkgs/development/tools/mongosh/default.nix:6]: So roughly how many unused packages (measured in disk space)? Having to manually generate a lock file would greatly increase the burden of maintenance.

File: pkgs/development/tools/mongosh/generate.sh [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
 
-node2nix \
-    --no-copy-node-env \
-    --node-env ../../node-packages/node-env.nix \
-    --input packages.json \
-    --output packages.nix \
-    --composition composition.nix \
-    --strip-optional-dependencies \
-    --nodejs-16
+mkdir mongosh
```
Comment (NickCao) [inline pkgs/development/tools/mongosh/generate.sh:6]: Use `mktemp -d` here.

**Review by NickCao — COMMENTED** [review]
And squash the commits.

**Pull request merged**